test/liken_test.exs

```diff
@@ -157,22 +181,81 @@ defmodule LikenTest do
   end
 
   test "a Struct is like the same Struct" do
-    assert %TestStruct{a: 1} |> like?(%TestStruct{a: 1})
+    expected = %TestStruct{a: 1}
+    actual   = %TestStruct{a: 1}
+    assert actual |> must_contain(expected)
   end
 
   test "a Struct is like a Struct with a subset of the attributes" do
-    assert %TestStruct{a: 1, b: 2} |> like?(%TestStruct{a: 1})
+    actual = %TestStruct{a: 1, b: 2}
+    expected = %{a: 1}
+
+    assert actual |> must_contain(TestStruct, expected)
   end
 
   test "a Struct is not like a Struct with a superset of the attributes" do
-    refute %TestStruct{a: 1} |> like?(%TestStruct{a: 1, b: 2})
+    actual = %TestStruct{a: 1}
+    expected = %{a: 1, b: 2}
+
+    func = fn -> assert actual |> must_contain(TestStruct, expected) end
+
+    assert_raise ExUnit.AssertionError, """
+    Actual value was not an exact match for Expected value:
+
+    At:
+    Map[:b]
+
+    Actual:
+    nil
+
+    Expected:
+    2
+
+    """, func
   end
 
   test "a Struct is not like a Struct with the same attributes but different struct types" do
-    refute %TestStruct{a: 1} |> like?(%FooStruct{a: 1})
+    actual = %TestStruct{a: 1}
+    expected = %{a: 1, b: 2}
+
+    func = fn -> assert actual |> must_contain(FooStruct, expected) end
+
+    assert_raise ExUnit.AssertionError, """
+    Actual value was not like Expected value:
+
+    Actual:
+    %LikenTest.TestStruct{a: 1, b: nil}
+
+    Expected:
+    %{a: 1, b: 2}
+
+    """, func
   end
 
+  test "the ultimate example" do
+    actual = %{
+      first: [1, 2, 3, %{us: "us", them: "them"}, 5],
+      second: "ignored",
+      third: %TestStruct{a: 1, b: 2},
+      fourth: [
+        %{key1: "111", key2: "222"},
+        %{key3: [3,3,3,3], key4: "444"}
+      ]
+    }
+    expected = %{
+      first: includes?([1, includes?(%{us: "us"})]),
+      third: includes?(TestStruct, %{a: 1}),
+      fourth: includes?([
+        includes?(%{key2: "222"}),
+        includes?(%{key3: includes?([3,3,3,3]), key4: "444"})
```

Comment:
So after our discussion, it turns out you DO need to do an `includes?` all the way down.  e.g. the array at `fourth` has two elements, and we list both of them, but without an `includes?` around the array, it doesn't work.  But WITH the `includes?` it does work.  So maybe we want to fix it.  Or maybe we live with it